src/coreclr/tools/Common/TypeSystem/Ecma/EcmaSignatureParser.cs

```diff
@@ -255,6 +255,8 @@ private TypeDesc ParseTypeImpl(SignatureTypeCode typeCode)
                         return _tsc.GetFunctionPointerType(sig);
                     else
                         return null;
+                case SignatureTypeCode.Sentinel:
+                    throw new TypeSystemException.BadImageFormatException();
```

Comment:
Based on the nuance captured in this PR's description, I think a comment here is warranted.

Reply:
It's possible that the owners for this area might want to change the default case to throw this exception instead of adding a new case statement, so I wanted to wait on adding a comment until they take a look.

Reply:
Thank you!

Yes, I would change the `default` case to throw the type system exception. Looks like this was missed when type system exceptions were introduced. Type system should only throw these.

Could you also do `ThrowHelpers.ThrowBadImageFormatException()` instead? We had configurations in the past where this code was also used at runtime and in that config the throw helper would throw the actual BadImageFormat exception, not the type system flavor.